test/gurka/test_disable_hierarchy.cc

```diff
@@ -0,0 +1,83 @@
+#include "gurka.h"
+#include "proto/options.pb.h"
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+// utils
+// Create costing options (reference: /test/astar.cc)
+void create_costing_options(Options& options, Costing::Type type) {
+  const rapidjson::Document doc;
+  sif::ParseCosting(doc, "/costing_options", options);
+  options.set_costing_type(type);
+}
+
+// Set disable_hierarchy_pruning to true in costing options
+void set_disable_hierarchy_pruning(Options& options, Costing::Type type) {
+  Costing* costing = &(*options.mutable_costings())[type];
+  auto* co = costing->mutable_options();
+  co->set_disable_hierarchy_pruning(true);
+}
+
+// Test the hierarchy limits are actually disabled when disable_hierarchy_pruning = true
+void doTest(Costing::Type costing_type) {
+  // Set costing options
+  Options options;
+  set_disable_hierarchy_pruning(options, costing_type);
+  create_costing_options(options, costing_type);
+  auto travel_mode = valhalla::sif::TravelMode::kDrive;
+  const auto mode_costing = valhalla::sif::CostFactory().CreateModeCosting(options, travel_mode);
+
+  // Check hierarchy limits
+  auto& hierarchy_limits =
+      mode_costing[int(travel_mode)]->GetHierarchyLimits(); // access mutable limits
+  for (auto& hierarchy : hierarchy_limits) {
+    EXPECT_EQ(hierarchy.max_up_transitions, kUnlimitedTransitions);
+  }
+}
+
+TEST(DisableHierarchyPruningTest, Auto) {
+  Costing::Type type = Costing::auto_;
+  doTest(type);
+}
+
+TEST(DisableHierarchyPruningTest, Bike) {
+  Costing::Type type = Costing::bicycle;
+  doTest(type);
+}
```

Comment:
bike, pedestrian & none you can remove, we're only looking for motorized vehicles as the others already have hierarchies disabled

extra points if we you want to make the others a parameterized test:) (e.g. gurka/test_avoids.cc does that)

Reply:
Oops I just wanted to have a try, forgot to remove them :D
Parameterized test is finished.